src/pcg/corridor_generator.cpp

```diff
@@ -0,0 +1,113 @@
+            fourth_connecting_point = second_middle.shifted_by(0, -second_room.get_height() / 2);            
+        }
+    } else {
+        if (first_middle.x > second_middle.x) {
+            fourth_connecting_point = first_middle.shifted_by(-first_room.get_width() / 2, 0);
+            first_connecting_point = second_middle.shifted_by(second_room.get_width() / 2, 0);
+        } else {
+            first_connecting_point = first_middle.shifted_by(first_room.get_width() / 2, 0);
+            fourth_connecting_point = second_middle.shifted_by(-second_room.get_width() / 2, 0);            
+        }
+    }
+            
+    Point midpoint((first_connecting_point.x + fourth_connecting_point.x) / 2, (first_connecting_point.y + fourth_connecting_point.y) / 2);
+    Point second_connecting_point(0, 0);
+    Point third_connecting_point(0, 0);
+
+    std::vector<BoundingBox> corridor_boxes;
+    corridor_boxes.reserve(3);
+
+    if (horizontal_corridor) {
+        second_connecting_point = Point(midpoint.x, first_connecting_point.y);
+        third_connecting_point = Point(midpoint.x, fourth_connecting_point.y);
+
+        corridor_boxes.emplace_back(first_connecting_point.shifted_by(0, -CORRIDOR_WIDTH/2), second_connecting_point.shifted_by(0, CORRIDOR_WIDTH/2));
+
+        if(midpoint.y < first_connecting_point.y)
+            corridor_boxes.emplace_back(third_connecting_point.shifted_by(-CORRIDOR_WIDTH/2, -CORRIDOR_WIDTH/2), second_connecting_point.shifted_by(CORRIDOR_WIDTH/2, CORRIDOR_WIDTH/2));
+        else
+            corridor_boxes.emplace_back(second_connecting_point.shifted_by(-CORRIDOR_WIDTH/2, -CORRIDOR_WIDTH/2), third_connecting_point.shifted_by(CORRIDOR_WIDTH/2, CORRIDOR_WIDTH/2));
+
+        corridor_boxes.emplace_back(third_connecting_point.shifted_by(-CORRIDOR_WIDTH/2, -CORRIDOR_WIDTH/2), fourth_connecting_point.shifted_by(0, CORRIDOR_WIDTH/2));
+            
+    } else {
+        second_connecting_point = Point(first_connecting_point.x, midpoint.y);
+        third_connecting_point = Point(fourth_connecting_point.x, midpoint.y);
+
+        corridor_boxes.emplace_back(first_connecting_point.shifted_by(-CORRIDOR_WIDTH/2, 0), second_connecting_point.shifted_by(CORRIDOR_WIDTH/2, 0));
+
+        if(midpoint.x < first_connecting_point.x)
+            corridor_boxes.emplace_back(third_connecting_point.shifted_by(-CORRIDOR_WIDTH/2, -CORRIDOR_WIDTH/2), second_connecting_point.shifted_by(CORRIDOR_WIDTH/2, CORRIDOR_WIDTH/2));
+        else
+            corridor_boxes.emplace_back(second_connecting_point.shifted_by(-CORRIDOR_WIDTH/2, -CORRIDOR_WIDTH/2), third_connecting_point.shifted_by(CORRIDOR_WIDTH/2, CORRIDOR_WIDTH/2));
+    
+        corridor_boxes.emplace_back(third_connecting_point.shifted_by(-CORRIDOR_WIDTH/2, 0), fourth_connecting_point.shifted_by(CORRIDOR_WIDTH/2, 0));
+    }
+
+    return corridor_boxes;
+}
+
+void CorridorGenerator::run(Scene& scene) {
```

Comment:
Try to organize cpp file, so the methods written above are using methods written below. It's easier to read, just like poem